### PR TITLE
Change target to tables format for Multivariate regressor.

### PR DESCRIPTION
**Pull request by ayush-1506**
Fixes: #11

### CODE REVIEW

**Comment by ayush-1506** [comment]
@ablaom I'm not sure why the test fails on travis; they pass locally. Any idea?

File: src/regressor.jl [hunk 1]
```diff
@@ -65,7 +65,7 @@ function collate(model::Union{NeuralNetworkRegressor, MultivariateNeuralNetworkR
     row_batches = Base.Iterators.partition(1:length(y), batch_size)
 
     Xmatrix = MLJModelInterface.matrix(X)'
-    if y isa AbstractVector{<:Tuple}
+    if y isa Tables.MatrixTable
         return [(Xmatrix[:, b], ymatrix[:, b]) for b in row_batches]
```
Comment (ablaom) [inline src/regressor.jl]: This should be `if Tables.istable(y)` not `if y isa Tables.MatrixTable`, which is too restrictive. For, examlple, user is allowed to send a DataFrame.

File: src/regressor.jl [hunk 2]
```diff
@@ -78,12 +78,17 @@ function MLJModelInterface.fit(model::Union{NeuralNetworkRegressor, Multivariate
 
     # When it has no categorical features
     n_input = Tables.schema(X).names |> length
-    n_output = length(y[1])
-    chain = fit(model.builder, n_input, n_output)
-
     data = collate(model, X, y, model.batch_size)
 
-    target_is_multivariate = y isa AbstractVector{<:Tuple}
+    target_is_multivariate = y isa Tables.MatrixTable
```
Comment (ablaom) [inline src/regressor.jl]: ditto here

File: src/regressor.jl [hunk 3]
```diff
@@ -78,12 +78,17 @@ function MLJModelInterface.fit(model::Union{NeuralNetworkRegressor, Multivariate
...
+    if target_is_multivariate
+        target_columns = Tables.schema(y).names
+    else
+        target_columns = [""] # We won't be using this
+    end
+
+    n_output = length(target_columns)
+    chain = fit(model.builder, n_input, n_output)
```
Comment (ablaom) [inline src/regressor.jl]: This gives n_ouput = 0 in the univariate case. Don't you want this to be 1?

File: src/regressor.jl [hunk 4]
```diff
@@ -137,7 +141,12 @@ function MLJModelInterface.update(model::Union{NeuralNetworkRegressor, Multivari
         epochs = model.epochs - old_model.epochs
     else
         n_input = Tables.schema(X).names |> length
-        n_output = length(y[1])
+        if target_is_multivariate
+            target_columns = Tables.schema(y).names
+        else
+            target_columns = [""] # We won't be using this
+        end
+        n_output = length(target_columns)
```
Comment (ablaom) [inline src/regressor.jl]: see earlier comment re n_output

File: src/regressor.jl (hunk 3 above)
Comment (ayush-1506) [reply src/regressor.jl]: @ablaom n_output will be 1 in this case, which is the length of [""].

**Comment by ablaom** [comment]
Sorry about the n_output comment. My mistake.

That is weird. Your PR is also passing locally for me.

**Comment by ablaom** [comment]
Is this a clue? It seems travis is picking an ancient version of MLJBase for testing. We are up to 0.12.1 now. 

Screen Shot 2020-03-16 at 2 42 17 PM

**Comment by ayush-1506** [comment]
I'll look into this. Not sure why it's picking up an older version. I could try pinning MLJBase to the required version, hopefully that should solve it. (Although I guess this shouldn't be the preferred method)

**Comment by ablaom** [comment]
FYI: As I understand it, [compat] declarations are ignored for packages not listed under [deps] (see https://github.com/JuliaLang/Pkg.jl/issues/1179) (although this may change in the future (https://github.com/JuliaLang/Pkg.jl/issues/1285)).

**Comment by ayush-1506** [comment]
@ablaom At the moment, all packages in [deps] are also declared in [compat]. How is that related to MLJBase?

**Comment by ayush-1506** [comment]
Also, what I'm unable to understand why travis passed on the previous PRs but fails for this. (I've not made any additional changes to Project.toml here, so it should naturally pull the same version of MLJBase (0.12.0) as in previous PRs, right?)

**Comment by ablaom** [comment]
> @ablaom At the moment, all packages in [deps] are also declared in [compat]. How is that related to MLJBase?

For new release requests to be fast-tracked (auto-merged into General.jl) it is now a requirement that all [deps] not in the standard library be given [compat] upper bounds. As to your question, " How is that related to MLJBase?" I'm not sure I understand what you mean. Of course the fact that we cannot cap versions of "test-only" packages is a flaw, but this is a bigger issue, as I point out above, and should not concern us here: As I understand it, testing ought to load the latest version of [targets] packages not creating conflicts with the [deps]/[compat] specification. In the case of MLJBase, this should be 0.12.1 (or thereabouts) and I have no idea why travis is not using this version.

> Also, what I'm unable to understand why travis passed on the previous PRs but fails for this. (
Sorry, me neither. Have you tried re-triggering travis with a trivial commit?

**Comment by ayush-1506** [comment]
Just tried that, still fails. @tlienart would be great if you could also have a look when you get time. Maybe there's something we might have overlooked. Thanks

**Comment by tlienart** [comment]
wrong compat bound for Tables, will fix

**Comment by tlienart** [comment]
Uh I don't have write access... 

Basically most important one is to bump the tables compat to `^1.0`. 

Notes: (at the risk of sounding annoying)

* please use Juno's  automatic formater to get rid of spurious whitelines and have more consistent code formatting
* please add docstrings
* please add comments apart for the super obvious places, assume someone else than you will have to maintain this

Also ideally please give Anthony and I write access (I'm assuming Anthony already has it). 

Otherwise it's great that MLJFlux is back on track!

**Comment by ayush-1506** [comment]
@tlienart Thanks for that. Unfortunately travis still pulls the older MLJBase.

**Comment by tlienart** [comment]
Make it an explicit dependency, add compat bounds and see what fails on travis

**Comment by ablaom** [comment]
Given @tlienart admin access

**Comment by ayush-1506** [comment]
@tlienart I tried a bunch of things, nothing seems to work. (Even locally, I deleted the manifest file and did a clean install and it pulls the older MLJBase). Not quite sure what's going wrong. Would be great if you could take a look too. Thanks.

**Comment by ablaom** [comment]
Okay, finally we are getting somewhere. By retroactively adding upper bounds to all versions of all MLJ repos, we get useful conflict error from MLJFlux travis test, suggesting MLJScientifictTypes is the culprit: It needs to update its ColorTypes [compat], which I am now working on.

**Pull request merged**